foxglove/extensions/system-status-panel/src/SystemStatusPanel.tsx

```diff
@@ -0,0 +1,121 @@
+import { PanelExtensionContext, RenderState, MessageEvent, Immutable } from "@foxglove/studio";
+import { Box, Typography } from "@mui/material";
+import Paper from "@mui/material/Paper";
+import Table from "@mui/material/Table";
+import TableBody from "@mui/material/TableBody";
+import TableCell, { tableCellClasses } from "@mui/material/TableCell";
+import TableContainer from "@mui/material/TableContainer";
+import TableRow from "@mui/material/TableRow";
+import { useTheme } from "@mui/material/styles";
+import { useEffect, useState } from "react";
+import { createRoot } from "react-dom/client";
+
+const SYSTEM_USAGE_TOPIC = "/system/usage"; // Topic to watch for system usage data
+
+interface SystemUsage {
+  cpu_percent: number;
+  cpu_speed: number;
+  gpu_percent: number;
+  gpu_speed: number;
+  gpu_memory: Memory;
+  ram: Memory;
+  disk: Memory;
+}
+interface Memory {
+  used: number;
+  total: number;
+  percentage: number;
+}
+
+function createData(status: string, value?: number) {
+  return { status, value };
+}
+
+type State = {
+  cpuUsage?: number;
+  ramUsage?: number;
+};
+
+function SystemStatusPanel({ context }: { context: PanelExtensionContext }): JSX.Element {
+  const [state, setState] = useState<State>({});
+
+  context.subscribe([{ topic: SYSTEM_USAGE_TOPIC }]);
+
+  // Define values in table
+  const rows = [createData("CPU", state.cpuUsage), createData("RAM", state.ramUsage)];
+
+  // Watch system usage topic and update state
+  useEffect(() => {
+    context.onRender = (renderState: Immutable<RenderState>, done) => {
+      // Reset state when the user seeks the video
+      if (renderState.didSeek ?? false) {
+        setState({});
+      }
+
+      // Update system usage state
+      if (renderState.currentFrame && renderState.currentFrame.length > 0) {
+        const latestFrame = renderState.currentFrame[renderState.currentFrame.length - 1] as MessageEvent<SystemUsage>;
```

Comment:
To get the last element in the `renderState.currentFrame` array, [`Array.at(-1)`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at) can be used.